json-endpoints/jetpack/class.jetpack-json-api-plugins-install-endpoint.php

```diff
@@ -58,18 +58,14 @@ protected function validate_plugins() {
 				return new WP_Error( 'plugin_already_installed', __( 'The plugin is already installed', 'jetpack' ) );
 			}
 
-			$response    = wp_remote_get( "http://api.wordpress.org/plugins/info/1.0/$slug" );
-			$plugin_data = unserialize( $response['body'] );
-			if ( is_wp_error( $plugin_data ) ) {
-				return $plugin_data;
-			}
-
-			$this->download_links[ $slug ] = $plugin_data->download_link;
-
 		}
 		return true;
 	}
 
+	protected static function generate_wordpress_org_plugin_download_link( $plugin_slug ) {
+		return "https://downloads.wordpress.org/plugin/{$plugin_slug}.latest-stable.zip";
```

Comment:
`$plugin_slug` needs to be stripped down to just the folder name, eg:

```
$plugin_id = substr( $plugin_slug, 0, strpos( $plugin_slug, '/' ) );
return "https://downloads.wordpress.org/plugin/{$plugin_id}.latest-stable.zip";
```

If we fix that, i'd say this is good to go.